Update index.html

dubinsky opened the pull request:
Made the tabs cleaner (spacing and color, etc), added a title and slogan in the proper place. Couldn't get the image to move from where it currently is. I can't figure out how to make <link> tag work since I can't seem to get the file path for the main.css file in our github. When I used <link src = "main.css"> it didn't work. If you know how to get the correct file path/name, that would be great :)

CriticalWasTaken committed:
Update index.html

Made the tabs cleaner (spacing and color, etc), added a title and slogan in the proper place. Couldn't get the image to move from where it currently is. I can't figure out how to make <link> tag work since I can't seem to get the file path for the main.css file in our github. When I used <link src = "main.css"> it didn't work. If you know how to get the correct file path/name, that would be great :)

dubinsky commented on index.html:
```diff
@@ -5,7 +5,7 @@
 <script>
 // JS for the tabs
-function openCity(evt, cityName) {
+function openCity(evt, tab) {
```
please rename this away from `...City`

CriticalWasTaken replied:
Sorry, missed this. Will fix in next pull request im'h,

dubinsky replied:
> Sorry, missed this. Will fix in next pull request im'h,

Please do it in THIS pull request, not the next one.

CriticalWasTaken replied:
Done.

dubinsky replied:
Please rename `openCity` to something that does not mention cities.

CriticalWasTaken replied:
Oh no! Am I editing the wrong branch? I fixed this issue, but not here??

dubinsky commented on index.html:
```diff
@@ -145,22 +122,25 @@ <h1 class = title>MATH WORLDS</h1>
-  <button class="tablinks" onclick="openCity(event, 'Home')">Home</button>
-  <button class="tablinks" onclick="openCity(event, 'AboutUs')">About Us</button>
-  <button class="tablinks" onclick="openCity(event, 'MWGames')">MW Games</button>
+  <button class="tablinks" onclick="openCity(event, 'Home')"><strong>Home</strong></button>
```
please do not use presentation tags like `<strong>`; style button labels using CSS.

CriticalWasTaken replied:
Will fix in the next pull request im'h.

CriticalWasTaken replied:
Done. Why is this? Purely convention?

dubinsky replied:
Hmm... "done" in what sence? Please DO NOT USE PRESENTATION TAGS like `strong`; style with CSS.

> Why is this?

To separate structure (HTML) from presentation (CSS).
This is also why CSS needs to be in a separate file.

CriticalWasTaken replied:
I did this, but was it in the wrong branch?

dubinsky commented on index.html:
```diff
@@ -191,7 +171,14 @@ <h3>MW Games</h3>
+	<p> Contact us at insert contact info here
+	</p>
 </div>
+	<img class = logo, src = "file:///C:/Users/The%20Critic/Downloads/Logo-Final-outlines.svg", width = 150px, height = 150vw, alt = "Happy Robot">
```
do not use absolute paths or paths that are only valid on your machine;
this needs to work when it is deployed in the cloud.
Use paths that are relative to the HTML file or absolute path from the root of the deployment.

CriticalWasTaken replied:
Sorry, but I'm not sure how to locate such an absolute path to do so.

dubinsky replied:
> Sorry, but I'm not sure how to locate such an absolute path to do so.

As I wrote above:

`main.css` has absolute path `/main.css` and relative to the `index.html` its path is `./main.css` or `main.css`.
When developing locally, do you use some kind of a web server to serve the pages you are working with?
If not, you need to start doing this, or paths above won't work on your machine...

CriticalWasTaken replied:
When developing locally, I haven't been using <link> to access the css from somewhere else. Maybe we can discuss this at some point, to make sure I don't mess this all up?

dubinsky replied:
To not mess this all up, you need to use a local web server while working on it. So, again: are you?

For explanation why it is needed and some suggestions how to go about it, see https://developer.mozilla.org/en-US/docs/Learn/Common_questions/set_up_a_local_testing_server

CriticalWasTaken replied:
I don't fully understand what this local testing server does or is supposed to do. Can we discuss this over Shabbos?

CriticalWasTaken replied:
So no, I don't think I'm currently using this.

dubinsky replied:
> I don't fully understand what this local testing server does or is supposed to do. Can we discuss this over Shabbos?

If you are allowed to discuss such topics on Shabbos - sure :)

dubinsky replied:
remove your-local-machine-specific paths

dubinsky commented:
> Couldn't get the image to move from where it currently is.
Not sure what you mean by that.

> I can't figure out how to make tag work since I can't seem to get the file path for the main.css file in our github.
> When I used it didn't work. If you know how to get the correct file path/name, that would be great :)
`main.css` has absolute path `/main.css` and relative to the `index.html` its path is `./main.css` or `main.css`.
When developing locally, do you use some kind of a web server to serve the pages you are working with?
If not, you need to start doing this, or paths above won't work on your machine...

CriticalWasTaken committed:
Update index.html

The only change should be that the css should be in a separate file, and linked so that it still affects the page. I think this needs to be implemented to allow me to see if it worked completely. Please let me know any errors on my end that need to be fixed.

CriticalWasTaken committed:
Update index.html

removed strong tags and renamed the openCity function

dubinsky commented on index.html:
```diff
@@ -145,22 +123,25 @@ <h1 class = title>MATH WORLDS</h1>
-  <button class="tablinks" onclick="openCity(event, 'AboutUs')">About Us</button>
-  <button class="tablinks" onclick="openCity(event, 'MWGames')">MW Games</button>
+  <button class="tablinks" onclick="openTab(event, 'Home')">Home</button>
+  <button class="tablinks" onclick="openTab(event, 'AboutUs')"About Us</button>
```
Some closing angle brackets ">" seem to be missing...

CriticalWasTaken committed:
Update index.html
